### DISCUSSION TITLE
Migrating LuaSec (also: enable Discussions?)

### DISCUSSION THREAD

**Discussion by hishamhm** [lunarmodules/.github#7] [General]
Hi,

Not sure this is the right place to post this but I don't think we have a Lunar Modules forum (should we enable org-wide Discussions? Looks like Github has that feature.)

@brunoos, LuaSec author, contacted me about moving LuaSec over to Lunar Modules. I'm a bit swamped right now so I can't help out too much with the process, but I thought I'd ping some of the people who have helped others in the past: @Tieske @alerque, can you guys provide some direction?

Thanks!!

**Comment by Tieske** [comment]
discussions enabled and issue migrated here

**Comment by Tieske** [comment]
Moving LuaSec should be fine. Since he owns the repo and the LuaRocks entry, it should be easy.

Only need to set up a team and add him to it I guess.

The one decision to make is whether it should stay under the current LuaRocks account or it should be moved under the LunarModules luarocks account. But that's up to @brunoos

**Reply by alerque** [reply]
Yup, second this. The LuaRocks publishing can use our org key to publish under Lunarmodules or their own API key to publish under the current account, it shouldn't matter much to setting things up here.

I can help some with wiring up the CI and such after the repo moves. For now the usual new team + inviting @brunoos to it is the first step, then they can make the transfer and we can tweak the permissions to be team based and start work on the CI workflows.

**Comment by Tieske** [comment]
Created a new team, and invited @brunoos. I think @brunoos should then be able to transfer the repo to lunarmodules.

@brunoos; before contacting @hishamhm , where did you check on how to get started with handing over to lunarmodules?
(asking, because that location is probably the place where we should add some instructions for other people as well to get started)

**Comment by brunoos** [comment]
Just a question: after transfer the repo, do the current and old rockspecs at luarocks have to be fixed in some way, since they point to my repo?

**Reply by alerque** [reply]
No, when you transfer a repository o GitHub it automatically adds redirects from the old location to the new. These work even for `git clone` operations and all the repository URLs.

The caveat is that if you ever then *fork* the new repository location back to your user then all those redirects die and start linking to your fork. After transferring the repository here, either work with it directly here (which as the owner you'll continue to be able to do) or carefully fork it to a different name in your namespace.

Future releases of rockspecs should of course use the new URL, bot the old ones as published should continue to work the way they do now.

**Reply by Tieske** [reply]
I think that if you enable the automation for rockspec uploads, you shoukld easily be able to create a new rockspec with a new revision only (so no new version). And it would be uploaded automatically.

Prerequitiste would be that either;
- the luarocks ownership is also transferred to lunarmodules (in which case the org-wide secret for the luarock upload api-key can be used), or
- you set your luarocks api-key as a secret specific to this repo

**Reply by brunoos** [reply]
I think changing the ownership is the best option. How do I do this?

**Reply by alerque** [reply]
@hishamhm can take care of that, and your comment here along with what he already has in hand should be enough authentication to go on. :wink:

**Reply by hishamhm** [reply]
The `luasec` entry is now transferred to the `lunarmodules` account at luarocks.org: https://luarocks.org/modules/lunarmodules/luasec

**Comment by brunoos** [comment]
The module was transferred.

**Reply by Tieske** [reply]
Access to the repo is now through the LuaSec team. Within that team you have the "maintainer" role.
